### PR TITLE
Allow enchants on fireworks/charges/potions/banners

### LINKED ISSUE #3879: Firework Stars do not receive enchants when obtained from ESSX kits or give command
### Information

**Full output of `/ess version`:**
Note, our main servers are running `2.17.2.96` where this is also a present issue (we plan to update in the very near future), more on that below. For sake of testing, I used old versions and latest from Spigot.
```
28.12 00:38:33 [Server] Server thread/INFO CONSOLE issued server command: /ess version
28.12 00:38:33 [Server] Server thread/INFO Server version: 1.16.4-R0.1-SNAPSHOT git-Spigot-a19903d-5b74714 (MC: 1.16.4)
28.12 00:38:33 [Server] Server thread/INFO EssentialsX version: 2.18.2.0
28.12 00:38:33 [Server] Server thread/INFO LuckPerms version: 5.2.35
28.12 00:38:33 [Server] Server thread/INFO PlaceholderAPI version: 2.10.9
28.12 00:38:33 [Server] Server thread/INFO Citizens version: 2.0.27-SNAPSHOT (build 2170)
28.12 00:38:33 [Server] Server thread/INFO CMI version: 8.7.12.2
28.12 00:38:33 [Server] Server thread/INFO Vault is not installed. Chat and permissions may not work.
28.12 00:38:33 [Server] Server thread/INFO You are running a server with limited API functionality. EssentialsX will still work, but certain features may be disabled.
```

**Server startup log:**
I removed the IP Address. Note minimal plugin usage is because this is a test server. This also happens on our main servers (all 6), as well as confirmed happen to other people via the ESSX #support channel 
- [latest.log](https://gist.github.com/GenSecrets/c7992d030cefa211aff9cb2746748431)

**EssentialsX config:**
Our main servers are on `2.17.2.96`, we do understand this is an incredibly old version, we have full plans to update once our events finish. The last few months of each year we have massive seasonal events and avoid any planned change during this time unless for emergency maintenance/update. For sake of testing, this is an issue on `2.17.2.96` as well as the current version `2.18.2` which I downloaded and used on the test server. Below are the freshly generated configs, no edits other than the testorb kit addition.
- [config.yml](https://gist.github.com/GenSecrets/d8f16691421873a56c98b2eb34ba47e8)
- [kits.yml](https://gist.github.com/GenSecrets/a9d87a43eccdb8ee49b085c1ac96b760)

### Details

**Description:**  

I'm relaying this bug report from the MOSS EssentialsX #support channel where Evidentsinger14#0001 advised me to open an issue and confirmed that this also happens on their server, something discovered during troubleshooting. 
- **Summary:** Enchantments are not placed on firework stars when received either from essentials "give" or essentials "kits" commands
- It does not matter if the console gives them, another plugin (aka console) gives them, or if a player runs the commands
- You however CAN enchant the fireworkstars if you hold the item in game and run the essentials "enchant" command
- If you run "/kitshow" it does display the enchantments, name, lore, amount, etc exactly as defined in config
- **There are no errors neither in game nor in the console**

This has worked in the past as we have used enchanted fireworkstars as "orbs" given from ESSX kits for years. It's a major piece of our server.

**Steps to reproduce:**  
1) Create a test kit, item: `fireworkstar` (or `firework_star`), any amount, any name, any lore
2) Add enchantments, ie unbreaking, efficiency, protection, etc to the kit. Any enchantment, any level, any amount
3) Gather kit in game
4) Copy the items line from your kit(without the dash), open MC, type `/give <username> <paste items line>` and receive the same result
5) Manually type out `/give <username> fireworkstar 1 unbreaking:1` and receive the same result
6) Attempt the same above with other items such as leather_chestplate, armorstand, etc and they WILL receive enchants. I did not exhaustively go through the MC item list and find out if there are other items. 

**Expected behavior:**  
- Enchantments placed on items properly as intended

**Console stack trace:**  
N/A

**Screenshots:**  
Command response and items received:
- [https://imgur.com/a/MxSCPaG](https://imgur.com/a/MxSCPaG)

### PR DESCRIPTION
**Pull request by imDaniX**
This PR fixes #3879
Also added `p` as a short name for `power` for fireworks meta.

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java [hunk 1]
```diff
@@ -226,28 +226,25 @@ public void addStringMeta(final CommandSource sender, final boolean allowUnsafe,
...
-            final int power = NumberUtil.isInt(split[1]) ? Integer.parseInt(split[1]) : 0;
-            final FireworkMeta meta = (FireworkMeta) stack.getItemMeta();
-            meta.setPower(power > 3 ? 4 : power);
-            stack.setItemMeta(meta);
         } else if (split.length > 1 && split[0].equalsIgnoreCase("itemflags") && hasMetaPermission(sender, "itemflags", false, true, ess)) {
             addItemFlags(string);
         } else if (MaterialUtil.isFirework(stack.getType())) {
-            //WARNING - Meta for fireworks will be ignored after this point.
-            addFireworkMeta(sender, false, string, ess);
+            if (split[0].equalsIgnoreCase("power") || !parseEnchantmentStrings(sender, allowUnsafe, split, ess)) {
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java]: Why the check for `power`? It isn't required in the firework meta

Comment (imDaniX) [reply Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java]: Moved lines 229-233 to addFireworkMeta method, so it is required now

Comment (JRoy) [reply Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java]: It shouldn't be though? We don't need the full meta to create a valid firework

Comment (imDaniX) [reply Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java]: This change does not affect the creation of fireworks, but rather makes it more consistent - this is the only action that belongs to fireworks meta, but isn't included in the method.
I might be just stupid, but I really see no reason for it not to be there

**Comment by JRoy** [comment]
Needs conflict resolving from latest commits (maybe my previous review comment should also make sense given my recent PR to fix some firework meta stuff)

**Review by JRoy — CHANGES_REQUESTED** [review]
While this *would* allow fireworks/potions/etc to parse enchantments, it would not, for example, parse a firework with firework meta **and** enchantment meta. If that can be added (maybe by splitting meta into a hashmap of tokens?)

**Comment by imDaniX** [comment]
> While this _would_ allow fireworks/potions/etc to parse enchantments, it would not, for example, parse a firework with firework meta **and** enchantment meta. If that can be added (maybe by splitting meta into a hashmap of tokens?)

Wait, but isn't `addStringMeta` method works word-by-word? This method can accept something like `unbreakable:1 effect:trail`, but it can't properly parse it anyway because it just [splits](https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java#L193) the string into 1-2 pieces(so it will become `unbreakable` and `1 effect:trail`), and even `parseStringMeta` calls the method for _each_ argument.

**Comment by JRoy** [comment]
> > While this _would_ allow fireworks/potions/etc to parse enchantments, it would not, for example, parse a firework with firework meta **and** enchantment meta. If that can be added (maybe by splitting meta into a hashmap of tokens?)
> 
> Wait, but isn't `addStringMeta` method works word-by-word? This method can accept something like `unbreakable:1 effect:trail`, but it can't properly parse it anyway because it just [splits](https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java?rgh-link-date=2021-01-28T16%3A28%3A28Z#L193) the string into 1-2 pieces(so it will become `unbreakable` and `1 effect:trail`), and even `parseStringMeta` calls the method for _each_ argument.

That should be fixed ideally then.

**Comment by imDaniX** [comment]
I mean, right now it works as-is. Here's the result of `/i fireworkrocket 1 efficiency:1 name:Angry_Creeper color:red fade:green type:creeper power:1 durability:1`
screenshot

**Review by JRoy — APPROVED** [review]

**Review by JRoy — APPROVED** [review]

**Pull request merged**